packages/expect/src/types.ts

```diff
@@ -67,6 +69,8 @@ export interface MatcherState {
   promise?: string;
   suppressedErrors: Array<Error>;
   testPath?: string;
+  matcherHintOptions?: MatcherHintOptions;
+  diffOptions?: DiffOptions;
```

Comment:
Please keep alphabetical order and please add them to a type test here:

https://github.com/facebook/jest/blob/22322c9acc3197e43337b5415755823f09f410fe/packages/expect/__typetests__/expect.test.ts#L47-L49

To run the test locally – build the library and run `yarn test-types`.